integration/read_write_mode_test.go

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+//go:build requires_docker
+// +build requires_docker
+
+package integration
+
+import (
+	"math"
+	"testing"
+	"time"
+
+	"github.com/grafana/e2e"
+	e2edb "github.com/grafana/e2e/db"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/prompb"
+	"github.com/stretchr/testify/require"
+
+	"github.com/grafana/mimir/integration/e2emimir"
+)
+
+func TestReadWriteMode(t *testing.T) {
+	s, err := e2e.NewScenario(networkName)
+	require.NoError(t, err)
+	defer s.Close()
+
+	minio := e2edb.NewMinio(9000, blocksBucketName, alertsBucketName, rulerBucketName)
+	require.NoError(t, s.StartAndWaitReady(minio))
+
+	flags := map[string]string{
+		"-common.storage.backend":              "s3",
+		"-common.storage.s3.access-key-id":     e2edb.MinioAccessKey,
+		"-common.storage.s3.secret-access-key": e2edb.MinioSecretKey,
+		"-common.storage.s3.endpoint":          minio.NetworkHTTPEndpoint(),
+		"-common.storage.s3.insecure":          "true",
+		"-blocks-storage.s3.bucket-name":       blocksBucketName,
+		"-alertmanager-storage.s3.bucket-name": alertsBucketName,
+		"-ruler-storage.s3.bucket-name":        rulerBucketName,
+		"-ingester.ring.replication-factor":    "1",
+		"-memberlist.join":                     "mimir-backend-1",
```

Comment:
Is using the backend instance as a seed node for the ring a sensible thing to do? Or is there a better option?